src/input.rs

```diff
@@ -463,7 +464,18 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         };
 
         for _ in 0..(line_delta.abs()) {
```

Comment:
This line_delta is always an integer value, so all decimals are cut off. This leads to problems with touch because the progress made torwards the last scroll position is lost. So you basically always start from 0 again which leads to some funky behavior.

In the current master the `self.ctx.mouse_mut().scroll_line` variable is modified to always store the decimal points with a `= to_scroll % 1.0` assignment. So when the method is hit for the next time, the scroll progress of the touchpad is preserved. I'd assume a similar approach might be appropriate here.